api/src/services/listing.service.ts

```diff
@@ -489,6 +489,148 @@ export class ListingService implements OnModuleInit {
     return result;
   }
 
+  /**
+   *
+   * @param listingId id for the listing we are querying for
+   * @param lang language that the translated listing should come back as
+   * @param view the form the listing should come back as (what data is returned)
+   * @returns an externalized version of the listing as a json object
+   */
+  async findOneAndExternalize(
+    listingId: string,
+    lang: LanguagesEnum = LanguagesEnum.en,
+    view: ListingViews = ListingViews.full,
+  ): Promise<string> {
+    const listing: any = await this.findOne(listingId, lang, view);
+    if (!listing) {
+      return JSON.stringify(listing);
```

Comment:
Do we need to stringify the listing here since it is either null or undefined. Could we just do `return "";`

Reply:
updated!